### ISSUE TITLE
[Question/Suggest] Session requests return

### ISSUE THREAD

**Issue by Rasko1** [psf/requests#5390]
As most of the people who will make requests to web sites will need to have the source, why don't you implement an optional parameter so when creating the session object we can set it to true so we have an attribute of the object that store the return of the last requests, with that we wouldn't need to create a new variable each time, that will save lot of times for coders and is easy to implement (from my point of view), I can do a PR if you think this suggestion is Good. Of course, I could modify my local version to do that but that wouldn't make any sense when sharing to others

**Comment by sigmavirus24** [comment]
It sounds like you're suggesting that a session store the last response on the session itself. In a single-threaded context, that's all well and good. In a multi-threaded/coroutine context this just flat out doesn't work.

**Comment by nateprewitt** [comment]
Yep, sigmavirus24 is correct. This brings a lot of issues in with shared state in a parallelized environment. I don't think we'll be able to implement this due to those constraints. Thanks for the suggestion though @Rasko1!
